plugins/node/opentelemetry-instrumentation-aws-lambda/test/integrations/lambda-handler.test.ts

```diff
@@ -541,6 +541,29 @@ describe('lambda handler', () => {
       assert.strictEqual(spans.length, 0);
     });
 
+    it('ignores sampled lambda context if env OTEL_LAMBDA_DISABLE_AWS_CONTEXT_PROPAGATION is set to "true"', async () => {
+      process.env['OTEL_LAMBDA_DISABLE_AWS_CONTEXT_PROPAGATION'] = 'true';
+      process.env[traceContextEnvironmentKey] = sampledAwsHeader;
+      initializeHandler('lambda-test/async.handler', {
+        disableAwsContextPropagation: true,
+      });
```

Comment:
This test sets both env variable and config object to the same value.
Since this PR adds the ability to set the option via env variable, I think it would be more interesting to test with an empty config object.

e.g. we should have a test that invokes 
```js
_config.disableAwsContextPropagation = true;
```

The current test will not enter the first `if` since:
```js
if (_config.disableAwsContextPropagation == null)
```
is not truthy

Reply:
I would even consider adding tests for some of the other new logic:
- config object has precedence over env (`disableAwsContextPropagation` set to false, `OTEL_LAMBDA_DISABLE_AWS_CONTEXT_PROPAGATION` set to "true" => false should be used).
- env variable set to something which is not string should be ignored (trigger the `typeof env['OTEL_LAMBDA_DISABLE_AWS_CONTEXT_PROPAGATION'] === 'string'` code
- env variable set to "True" or "TRUE" should be used.

Reply:
e2da529b